python/jupyterlab-chat/jupyterlab_chat/ychat.py

```diff
@@ -52,105 +54,105 @@ def yusers(self) -> Map:
         """
         return next(
-            ((msg, i) for i, msg in enumerate(self.get_messages()) if msg["id"] == id),
+            ((msg, i) for i, msg in enumerate(self.get_messages()) if msg.id == id),
             (None, None)
         )
 
-    def get_messages(self) -> list[dict]:
+    def get_messages(self) -> list[Message]:
         """
         Returns the messages of the document.
         :return: Document's messages.
         """
         return self._ymessages.to_py() or []
 
-    def add_message(self, message: dict) -> int:
+    def add_message(self, message: Message) -> int:
         """
         Append a message to the document.
         """
         timestamp: float = time.time()
-        message["time"] = timestamp
+        message.time = timestamp
         with self._ydoc.transaction():
-            index = len(self._ymessages) - next((i for i, v in enumerate(self.get_messages()[::-1]) if v["time"] < timestamp), len(self._ymessages))
+            index = len(self._ymessages) - next((i for i, v in enumerate(self.get_messages()[::-1]) if v.time < timestamp), len(self._ymessages))
             self._ymessages.insert(index, message)
             return index
 
-    def update_message(self, message: dict, index: int, append: bool = False):
+    def update_message(self, message: Message, index: int, append: bool = False):
         """
         Update a message of the document.
         If append is True, the content will be append to the previous content.
         """
         with self._ydoc.transaction():
-            initial_message = self._ymessages.pop(index)
+            initial_message: Message = self._ymessages.pop(index)
             if append:
-                message["body"] = initial_message["body"] + message["body"]
+                message.body = initial_message.body + message.body
             self._ymessages.insert(index, message)
 
-    def set_message(self, message: dict, index: Optional[int] = None, append: bool = False) -> int:
+    def set_message(self, message: Message, index: Optional[int] = None, append: bool = False) -> int:
         """
         Update or append a message.
         """
 
-        initial_message: Optional[dict] = None
+        initial_message: Optional[Message] = None
         if index is not None and 0 <= index < len(self._ymessages):
             initial_message = self.get_messages()[index]
         else:
             return self.add_message(message)
 
-        if not initial_message["id"] == message["id"]:
-            initial_message, index = self.get_message(message["id"])
+        if not initial_message.id == message.id:
+            initial_message, index = self.get_message(message.id)
             if index is None:
                 return self.add_message(message)
 
         self.update_message(message, index, append)
         return index
 
-    def get_single_metadata(self, name) -> dict:
+    def get_single_metadata(self, name: str) -> dict:
```

Comment:
I searched the code and found that this method isn't called anywhere else. Since developers can already safely access a record in the metadata via `ychat.get_metadata().get(key, None)`, can this method be removed?